kpatch/kpatch

```diff
@@ -265,6 +270,31 @@ wait_for_patch_transition() {
 	return 1
 }
 
+module_ref_count() {
+	local modname="$1"
+	[[ $(cat "/sys/module/$modname/refcnt" 2>/dev/null) != "0" ]]
+}
+
+wait_for_zero_module_ref_count() {
+	local modname="$1"
+	local i=0
+
+	# We can't rely on a zero refcount with kpatch.ko as it
+	# implements KPATCH_FORCE_UNSAFE with an additional reference on
+	# kpatch-patch modules to avoid potential crashes.
+	kpatch_core_loaded && return 0
+
+	module_ref_count "$modname" || return 0
+
+	echo "waiting (up to $MODULE_REF_WAIT seconds) for module refcount..."
```

Comment:
Does this usually happen instantaneously?  Just wondering if it would be better to print a message only if it needs to sleep.

Reply:
The way `wait_for_zero_module_ref_count()` is implemented is that it first checks the refcount and 99.99% of the time it will exit early, only printing the message if it needs to sleep.  Suggestions on renaming `module_ref_count()` if that might make the pattern clearer?

Reply:
Oh, I just glossed over line 287.  Looks good then!